miku-server/src/main/scala/org/http4s/miku/MikuHandler.scala

```diff
@@ -0,0 +1,253 @@
+package org.http4s.miku
+
+import java.io.IOException
+import java.time.{Instant, ZoneId}
+import java.time.format.DateTimeFormatter
+import java.util.Locale
+import java.util.concurrent.atomic.AtomicLong
+
+import cats.effect.{Async, Effect, IO}
+import cats.syntax.all._
+import io.netty.channel.{ChannelInboundHandlerAdapter, _}
+import io.netty.handler.codec.TooLongFrameException
+import io.netty.handler.codec.http._
+import io.netty.handler.timeout.IdleStateEvent
+import org.http4s.server.ServiceErrorHandler
+import org.http4s.util.execution.trampoline
+import org.http4s.{HttpService, Request, Response}
+import org.log4s.getLogger
+
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.{Failure, Success}
+
+/** Netty request handler
+  *
+  * Adapted from PlayRequestHandler.scala
+  * in
+  * https://github.com/playframework/playframework/blob/master/framework/src/play-netty-server
+  *
+  * Variables inside this handler are essentially local to a thread in the
+  * MultithreadedEventLoopGroup, as they are not mutated anywhere else.
+  *
+  * A note about "lastResponseSent" to help understand:
+  * By reassigning the variable with a `flatMap` (which doesn't require synchronization at all, since
+  * you can consider this handler essentially single threaded), this means that, we can run the
+  * `handle` action asynchronously by forking it into a thread in `handle`, all the while
+  * ensuring in-order writes for the handler thread by attaching the callback using `flatMap`.
+  * This ensures we can schedule more work asynchronously by streaming `lastResponseSent` like a
+  * FIFO asynchronous queue.
+  *
+  */
+sealed abstract class MikuHandler[F[_]](
```

Comment:
If you make this package private, it's easier to change it in a binary compatible fashion when the bugs start rolling in.

Reply:
That one I meant to do. I forgot. will change